std/os/bits/linux.zig

```diff
@@ -9,9 +9,12 @@ pub usingnamespace switch (builtin.arch) {
     .aarch64 => @import("linux/arm64.zig"),
     .arm => @import("linux/arm-eabi.zig"),
     .riscv64 => @import("linux/riscv64.zig"),
+    .mipsel => @import("linux/mipsel.zig"),
     else => struct {},
 };
 
+const is_mips = builtin.arch == .mipsel;
```

Comment:
I'd like this file follow the existing pattern, where it has only everything that is 100% consistent between architectures. Other files have arch-specific bits:

```
$ ls std/os/bits/linux
arm64.zig  arm-eabi.zig  errno.zig  riscv64.zig  x86_64.zig
```

I don't consider this a merge-blocker though. I'd be happy to do this chore myself.

I'd also be OK with a `@hasDecl` based approach where a declaration checks for an arch-specific override and falls back to a "default".

Reply:
MIPS is the weird kid on the block (together with PPC) where the ABI is slightly different here and there. Even the errno codes are different (hence the test failure, the `89` errno you see is `getrandom` returning `ENOSYS`) so we need to come up with a plan to avoid scattering the definitions here and there.

I'm fine with the `@hasDecl` approach, but #3316 stops me from using that pattern.

Speaking of errnos, I think the wisest approach is to have a `errno-generic.zig` and a `errno-mips.zig` (and eventually a `errno-mips64.zig` and `errno-ppc64.zig` and so on so forth).